src/app/components/ATIAnalytics/amp/index.test.tsx

```diff
@@ -27,4 +28,69 @@ describe('Amp ATI Analytics', () => {
+          producer: 'GAHUZA',
+          additionalProperties: {
+            app_name: 'news-gahuza',
+            app_type: 'amp',
+            content_language: 'rw',
+            product_platform: null,
+            // eslint-disable-next-line no-template-curly-in-string
+            referrer_url: '${documentReferrer}',
+            // eslint-disable-next-line no-template-curly-in-string
+            x5: '${sourceUrl}',
+            x8: 'simorgh',
+            x9: "US%20irashishikariza%20u%20Rwanda%20na%20DR%20Congo%20kugera%20ku%20mahoro%20bikazana%20n'ishoramari%20rya%20miliyari%20z'amadorari",
+            x10: null,
+            x11: '2025-05-02T07:00:25.419Z',
+            x12: '2025-05-02T07:00:25.419Z',
+            x13: 'Democratic+Republic+of+Congo~Rwanda~March+23+Movement~M23+offensive',
+            x14: '3548f44f-46f5-4e6e-8628-3f668f161691~8125f2a9-3259-4f35-ab75-d9a6577fdc88~b03e7bfd-9a46-4053-aeed-f9f55f5e5567~c78c7532-43b3-490d-ad5b-0fc47b906e42',
+            x16: '',
+            x17: 'Democratic+Republic+of+Congo~Rwanda~March+23+Movement~M23+offensive',
+            x18: null,
+          },
+        },
+        user: {
+          isSignedIn: false,
+        },
+      },
+      eventDetails: {
+        eventName:
+          'pageView' as ReverbBeaconConfig['eventDetails']['eventName'],
+      },
+    };
+
+    const { container } = render(
+      <AmpATIAnalytics
+        pageviewParams="key1=value1&key2=value2"
+        reverbParams={mockReverbParams}
+      />,
+    );
+
+    const ampAnalyticsComponents = container.querySelectorAll(
+      'amp-analytics script[type="application/json"]',
+    );
+    const ampAnalyticsConfiguration = ampAnalyticsComponents[0]?.innerHTML;
+
+    const {
+      requests: { base, pageview },
+    } = JSON.parse(ampAnalyticsConfiguration);
+
+    expect(ampAnalyticsComponents.length).toEqual(1);
+    expect(base).toEqual('https://logws1363.ati-host.net/hit.xiti?');
+    expect(pageview).toEqual(
+      // eslint-disable-next-line no-template-curly-in-string
+      "${base}s=598343&s2=40&p=gahuza.articles.cly1jw4x585o.page&r=${screenWidth}x${screenHeight}x${screenColorDepth}&re=${availableScreenWidth}x${availableScreenHeight}&hl=${timestamp}&lng=${browserLanguage}&x1=[urn%3Abbc%3Aoptimo%3Aasset%3Acly1jw4x585o]&x2=[amp]&x3=[news-gahuza]&x4=[rw]&x5=[${sourceUrl}]&x6=[${documentReferrer}]&x7=[article]&x8=[simorgh]&x9=[US%2520irashishikariza%2520u%2520Rwanda%2520na%2520DR%2520Congo%2520kugera%2520ku%2520mahoro%2520bikazana%2520n'ishoramari%2520rya%2520miliyari%2520z'amadorari]&x11=[2025-05-02T07%3A00%3A25.419Z]&x12=[2025-05-02T07%3A00%3A25.419Z]&x13=[Democratic%2BRepublic%2Bof%2BCongo~Rwanda~March%2B23%2BMovement~M23%2Boffensive]&x14=[3548f44f-46f5-4e6e-8628-3f668f161691~8125f2a9-3259-4f35-ab75-d9a6577fdc88~b03e7bfd-9a46-4053-aeed-f9f55f5e5567~c78c7532-43b3-490d-ad5b-0fc47b906e42]&x17=[Democratic%2BRepublic%2Bof%2BCongo~Rwanda~March%2B23%2BMovement~M23%2Boffensive]&ref=${documentReferrer}",
```

Comment:
Could we try to make this more readable? Either: 

- https://github.com/bbc/simorgh/blob/latest/src/app/components/ATIAnalytics/atiUrl/index.test.ts#L274-L287
- or something similar to https://github.com/bbc/simorgh/blob/latest/src/integration/common/analytics.amp.js#L17-L32

Reply:
I'll give `splitUrl` a try and report back.

`searchParams` provided by the instance of `URL` may not be a good
fit here. This is because `searchParams` decodes the query parameters.

By using `searchParams` in a unit test, we may not be able to catch
incorrectly formatted strings such as `x6=[%24%7BdocumentReferrer%7D]`,
which was being returned by `reverb-url-helper`.

See [fix PR](https://github.com/bbc/reverb-url-helper/pull/23) for reference.